sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala

```diff
@@ -113,6 +88,7 @@ object AliasIdentifier {
  */
 case class TableIdentifier(table: String, database: Option[String], catalog: Option[String])
   extends CatalystIdentifier {
+  assert(catalog.isEmpty || database.isDefined)
```

Comment:
This requires `database` must be defined if `catalog` is defined?

Reply:
yea, it's super weird of an identifier has catalog field but not database.